File: 11_Evaluation_of_Postfix_Expression.py
def cal(a,b,operator):
    if operator=='+':
        return int(a)+int(b)
    elif operator=='-':
        return int(a)-int(b)
    elif operator=='*':
        return int(a)*int(b)
    elif operator=='/':
        return int(a)/int(b)
        
class Solution:
    
    #Function to evaluate a postfix expression.
    def evaluatePostfix(self, S):
        operator=['*','/','+','-']
        temp=[]
        for i in S:
            if i not in operator:
                temp.append(i)
            else:
                if len(temp)>0:
                    temp1=temp.pop()
                    temp2=temp.pop()
                    res= cal(temp2,temp1,i)
                    temp.append(res)
        return temp[0]

File: 11_Evaluation_of_Postfix_Expression.cpp
class Solution
{
    public:
    //Function to evaluate a postfix expression.
    int evaluatePostfix(string S)
    {
        int n = S.length();
        int res;
        int a,b;
        stack<int> temp;
        for (int i=0;i<n;i++){
            if ((S[i]=='+' || S[i]=='-' || S[i]=='*' || S[i]=='/') and temp.size()>0){
                a=temp.top();
                temp.pop();
                b=temp.top();
                temp.pop();
                if(S[i]=='+')
                    res = int(a)+int(b);
                else if (S[i]=='-')
                    res = int(b)-int(a);
                else if (S[i]=='*')
                    res = int(a)*int(b);
                else if (S[i]=='/')
                    res = int(b)/int(a);
                temp.push(res);
            }
            else {
                temp.push(int(S[i]-'0'));
            }
        }
        return temp.top();
    }
};
